source/indices.d.ts

```diff
@@ -0,0 +1,21 @@
+/**
+Create a tuple type of an array's indices.
+
+@example
+```
+type FruitIndices = Indices<['apple', 'banana', 'plum']> // [0, 1, 2]
+```
+
+Can be used to create an index union (`0 | 1 | 2 | ...`)
+
+@example
+```
+type FruitIndex = Indices<['apple', 'banana', 'plum']>[number] // 0 | 1 | 2
+```
```

Comment:
Would be great to show a practical example of `FruitIndex` in use. I feel the example is not complete without.